lib/index.js

```diff
@@ -236,16 +267,130 @@ module.exports = class OtpRunner {
   }
 
   /**
-   * Builds a graph using OTP
+   * Recreate certain directories to make sure we're starting fresh
+   */
+  async clearFoldersAndFiles () {
+    try {
+      await fs.remove(this.getTargetFolder())
+      await fs.mkdirp(this.getTargetFolder())
+    } catch (e) {
+      this.log.error(e)
+      await this.fail('Failed to recreate graph router folder')
+    }
+  }
+
+  /**
+   * Returns the appropriate Graph.obj filename that OTP outputs after building
+   * a graph.
+   */
+  getGraphObjName () {
+    return this.isUsingOtp2x() ? 'graph.obj' : 'Graph.obj'
+  }
+
+  /**
+   * Downloads all needed files as specified in the manifest.
+   */
+  async downloadFiles () {
+    const downloadTasks = []
+
+    // add task to download OTP jar
+    downloadTasks.push({
+      dest: this.manifest.jarFile,
+      url: this.manifest.jarUrl
+    })
+
+    // add tasks to download files to the base folder
+    if (
+      this.manifest.baseFolderDownloads &&
+        this.manifest.baseFolderDownloads.length > 0
+    ) {
+      this.manifest.baseFolderDownloads.forEach(url => {
+        const splitUrl = url.split('/')
```

Comment:
This split happens twice. Might be easier to read if you just replace it with a util method `getLastPart` or something similar:

```javascript
          dest: path.join(
            this.getTargetFolder(),
            getLastPart(url)
          ),
```

Reply:
Refactored similar to as suggested when gtfsDownloads was removed.